frontend/src/views/applications/FacilityDetailsView.vue

```diff
@@ -87,15 +134,84 @@ export default {
       },
     },
     save: {
-      handler() {
-        this.$refs.form?.validate()
+      async handler() {
+        await this.saveApplication(true)
       },
     },
     next: {
       handler() {
         this.$router.push({ name: 'licences', params: { applicationGuid: this.$route.params.applicationGuid } })
       },
     },
+    primaryContact: {
+      handler() {
+        if (this.secondaryContact?.contactId != this.primaryContact?.contactId) return
+        this.secondaryContact = undefined
+      },
+    },
+  },
+  async created() {
+    await this.loadData()
+    this.primaryContact = this.contacts?.find((contact) => contact.contactId === this.currentApplication?.primaryContactId)
+    this.secondaryContact = this.contacts?.find((contact) => contact.contactId === this.currentApplication?.secondaryContactId)
+  },
+  methods: {
+    ...mapActions(useApplicationsStore, ['getApplication', 'setIsFacilityDetailsComplete']),
+
+    async loadData() {
+      try {
+        this.loading = true
+        await Promise.all([this.getFacilityInfo(), this.getContacts()])
+      } finally {
+        this.loading = false
+      }
+    },
+
+    async getContacts() {
+      try {
+        this.contacts = await FacilityService.getContacts(this.currentApplication?.facilityId)
+        this.contacts?.forEach((contact) => {
+          contact.fullName = contact.firstName + ' ' + contact.lastName
+          contact.roleName = this.getRoleNameById(contact.role)
+        })
+      } catch (error) {
+        this.setFailureAlert('Failed to get contacts for facilityId = ' + this.currentApplication?.facilityId, error)
+      }
+    },
+
+    async getFacilityInfo() {
+      try {
+        this.facilityInfo = await FacilityService.getFacility(this.currentApplication?.facilityId)
+      } catch (error) {
+        this.setFailureAlert('Failed to get Facility information for facilityId = ' + this.currentApplication?.facilityId, error)
+      }
+    },
+
+    getRoleNameById(roleId) {
+      const appStore = useAppStore()
+      return appStore.getRoleNameById(Number(roleId))
```

Comment:
I mentioned to Johnathan that you should be able to use ...mapState(useAppStore, ['getRoleNameById'] instead and avoid this extra code. Try it out.

Reply:
updated. FYI, I've only made this change in the FacilityDetails page. I haven't updated Johnathan's pages since I guess he may already updated them in his branch.